### PR TITLE
Mask password in data_path in the system.distribution_queue

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Mask password in data_path in the system.distribution_queue

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — COMMENTED** [review]
Fast test requires explicit linking of configs.
Other tests also.

**Comment by azat** [comment]
>Fast test requires explicit linking of configs.

Thanks! By some reason I though that `programs/server/config.d` is always used in tests.

**Comment by azat** [comment]
>Functional stateless tests (thread) — fail: 1, passed: 2408, skipped: 14

00840_long_concurrent_select_and_drop_deadlock flacky

File: src/Storages/System/StorageSystemDistributionQueue.cpp [hunk 1]
```diff
@@ -10,6 +10,75 @@
...
+    if (pw_start > user_pw_end)
+    {
+        /// No password in path
+        return path;
+    }
+    ++pw_start;
+
+    size_t pw_length = user_pw_end - pw_start;
+    /// Replace with a single '*' to hide even the password length.
+    masked_path.replace(pw_start, pw_length, 1, '*');
```
Comment (filimonov) [inline src/Storages/System/StorageSystemDistributionQueue.cpp:74]: IRL path usually refers to 2/3 nodes (replicas)
And the password is repeated there several times

**Review by filimonov — CHANGES_REQUESTED** [review]

File: src/Storages/System/StorageSystemDistributionQueue.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Storages/System/StorageSystemDistributionQueue.cpp:74]: Yes, we missed that.

Comment (azat) [reply src/Storages/System/StorageSystemDistributionQueue.cpp:74]: Indeed, under internal_replication=true path will contains all nodes, @filimonov nice catch!

Comment (azat) [reply src/Storages/System/StorageSystemDistributionQueue.cpp]: Fixed, test expanded

**Pull request merged**